### PR TITLE
Search by title

### PR DESCRIPTION
**Pull request by donpaul999**
Users can search recipes by the title.

- An index that stores documents, search and returns the first 10 matching recipes. 
 Matching recipes are sorted in ascending order by the title
- If there is no matching recipes we display a message:
` Right now we don't have any recipes matching your request. Stay close!`

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/SearchServlet.java [hunk 1]
```diff
@@ -26,25 +42,73 @@
...
 
-    ArrayList<Recipe> recipes = new ArrayList<>();
-    for (int i = 0; i < 5; ++i) {
       Recipe recipe = new Recipe();
-      recipe.setId(i);
-      recipe.setName("Cake");
-      recipe.setDescription("Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.");
-      recipes.add(recipe);
+      String description =
+        "Lorem quam dolor dapibus ante, sit amet pellentesque turpis lacus eu ipsum. Duis quis mi ut tortor interdum efficitur quis at mi. Pellentesque quis mauris vel ligula commodo scelerisque. In vulputate quam nisl, vel sagittis ipsum molestie quis. Suspendisse quis ipsum a sem aliquam euismod mattis sed metus.";
```
Comment (bessmarche) [inline src/main/java/com/google/sps/servlets/SearchServlet.java]: [OPTIONAL] Change the description to some info about Piece of Cake/ Just output some of the steps.

File: src/main/webapp/one-off/databaseInit.java [hunk 2]
```diff
@@ -14,7 +13,7 @@
 //
 //  If you have errors relating to 'isBlank()' try the following:
 //  + add this to the imports "import org.apache.commons.lang3.StringUtils.isBlank"
-//  + add this dependncy to the pom.xl: 
+//  + add this dependncy to the pom.xl:
```
Comment (bessmarche) [inline src/main/webapp/one-off/databaseInit.java]: change to pom.xml

File: src/main/webapp/one-off/databaseInit.java [hunk 3]
```diff
@@ -40,104 +39,124 @@
...
-        datastore.put(recipeEntity);
-    }
+    Entity recipeEntity = new Entity(keyRange.getStart());
+
+    recipeEntity.setProperty("title", title);
+    recipeEntity.setProperty("index_title", title.toLowerCase());
+    recipeEntity.setProperty("imgURL", imgURL);
+    recipeEntity.setProperty("ingredients", ingredients);
+    recipeEntity.setProperty("stepList", stepList);
+    recipeEntity.setProperty("author", "Piece of Cake");
```
Comment (bessmarche) [inline src/main/webapp/one-off/databaseInit.java]: Can you add the following properties: prep_time, cook_time, difficulty

File: src/main/java/com/google/sps/servlets/SearchServlet.java (hunk 1 above)
Comment (donpaul999) [reply src/main/java/com/google/sps/servlets/SearchServlet.java]: We will have a column for the description inside the database.

**Review by bessmarche — APPROVED** [review]

File: src/main/webapp/one-off/databaseInit.java (hunk 3 above)
Comment (donpaul999) [reply src/main/webapp/one-off/databaseInit.java]: This conflict will be solved when we are merging to the master, your PR already has this. #7

File: src/main/java/com/google/sps/servlets/Recipe.java [hunk 4]
```diff
@@ -13,10 +31,18 @@ public String getDescription() {
...
   }
 
-  public int getId() {
+  public long getId() {
     return id;
   }
 
+  public Double getDistance() {
+    return distance;
+  }
```
Comment (ramaroberto) [inline src/main/java/com/google/sps/servlets/Recipe.java]: Distance to what? Why is this part of the recipe?

File: src/main/java/com/google/sps/servlets/Recipe.java [hunk 5]
```diff
@@ -1,9 +1,27 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
 package com.google.sps.servlets;
 
-public class Recipe {
-  private int id;
+
+public class Recipe{
```
Comment (ramaroberto) [inline src/main/java/com/google/sps/servlets/Recipe.java]: All the public methods should contain a javadoc comment.

File: src/main/java/com/google/sps/servlets/Recipe.java [hunk 6]
```diff
@@ -25,7 +51,15 @@ public void setDescription(String givenDesc) {
...
   }
 
-  public void setId(int givenId) {
+  public void setId(long givenId) {
     id = givenId;
   }
+
+  public void setDistance(Double givenDistance) {
+    distance = givenDistance;
+  }
```
Comment (ramaroberto) [inline src/main/java/com/google/sps/servlets/Recipe.java]: I think this is no longer needed.

File: src/main/java/com/google/sps/servlets/SearchServlet.java [hunk 7]
```diff
@@ -26,25 +43,83 @@
...
-    request.setAttribute("messages", request.getParameter("searchterm"));
-
-    ArrayList<Recipe> recipes = new ArrayList<>();
-    for (int i = 0; i < 5; ++i) {
-      Recipe recipe = new Recipe();
-      recipe.setId(i);
-      recipe.setName("Cake");
-      recipe.setDescription("Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.");
-      recipes.add(recipe);
+    int error = 0;
```
Comment (ramaroberto) [inline src/main/java/com/google/sps/servlets/SearchServlet.java]: What's the use of this variable?

File: src/main/java/com/google/sps/servlets/SearchServlet.java [hunk 8]
```diff
@@ -26,25 +43,83 @@
...
+          "Lorem quam dolor dapibus ante, sit amet pellentesque turpis lacus eu ipsum. Duis quis mi ut tortor interdum efficitur quis at mi. Pellentesque quis mauris vel ligula commodo scelerisque. In vulputate quam nisl, vel sagittis ipsum molestie quis. Suspendisse quis ipsum a sem aliquam euismod mattis sed metus.";
+
+        recipe.setId(Long.parseLong(id));
+        recipe.setName(name);
+        recipe.setImage((String) recipeEntity.getProperty("imgURL"));
+        recipe.setDescription(description);
+
+        recipesList.add(recipe);
+      } catch (EntityNotFoundException e) {
+        error = 1;
```
Comment (ramaroberto) [inline src/main/java/com/google/sps/servlets/SearchServlet.java]: You should throw an exception if this is not supposed to happen.

File: src/main/java/com/google/sps/servlets/SearchServlet.java [hunk 9]
```diff
@@ -26,25 +43,83 @@
...
-    for (int i = 0; i < 5; ++i) {
-      Recipe recipe = new Recipe();
-      recipe.setId(i);
-      recipe.setName("Cake");
-      recipe.setDescription("Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.");
-      recipes.add(recipe);
+    int error = 0;
+    List<Recipe> recipesList = new ArrayList<>();
+    String titleToMatch = request.getParameter("searchterm");
+    Index index = getIndex();
```
Comment (ramaroberto) [inline src/main/java/com/google/sps/servlets/SearchServlet.java]: searchIndex, don't use generic variable names

File: src/main/java/com/google/sps/servlets/SearchServlet.java [hunk 10]
```diff
@@ -26,25 +43,83 @@
...
+              .setDirection(SortExpression.SortDirection.ASCENDING)
+              .setDefaultValue("")
+          )
+      )
+      .build();
+
+    Query query = Query
+      .newBuilder()
+      .setOptions(options)
+      .build(titleToMatch.toLowerCase());
```
Comment (ramaroberto) [inline src/main/java/com/google/sps/servlets/SearchServlet.java]: Please modularize this in a function called buildQuery

File: src/main/java/com/google/sps/servlets/SearchServlet.java [hunk 11]
```diff
@@ -26,25 +43,83 @@
...
+        String name = (String) recipeEntity.getProperty("title");
+
+        Recipe recipe = new Recipe();
+        String description =
+          "Lorem quam dolor dapibus ante, sit amet pellentesque turpis lacus eu ipsum. Duis quis mi ut tortor interdum efficitur quis at mi. Pellentesque quis mauris vel ligula commodo scelerisque. In vulputate quam nisl, vel sagittis ipsum molestie quis. Suspendisse quis ipsum a sem aliquam euismod mattis sed metus.";
+
+        recipe.setId(Long.parseLong(id));
+        recipe.setName(name);
+        recipe.setImage((String) recipeEntity.getProperty("imgURL"));
+        recipe.setDescription(description);
```
Comment (ramaroberto) [inline src/main/java/com/google/sps/servlets/SearchServlet.java]: Please modularize this into a function called "buildRecipe"

File: src/main/java/com/google/sps/servlets/SearchServlet.java [hunk 12]
```diff
@@ -26,25 +43,83 @@
...
+          )
+      )
+      .build();
+
+    Query query = Query
+      .newBuilder()
+      .setOptions(options)
+      .build(titleToMatch.toLowerCase());
+
+    Results<ScoredDocument> results = index.search(query);
```
Comment (ramaroberto) [inline src/main/java/com/google/sps/servlets/SearchServlet.java]: resultsByTitle

File: src/main/java/com/google/sps/servlets/SearchServlet.java [hunk 13]
```diff
@@ -26,25 +43,83 @@
...
+
+    for (ScoredDocument entity : results) {
+      try {
+        String id = entity.getId();
+        Entity recipeEntity = datastore.get(
+          KeyFactory.createKey("Recipe", Long.parseLong(id))
+        );
```
Comment (ramaroberto) [inline src/main/java/com/google/sps/servlets/SearchServlet.java]: Please modularize this into a function called getRecipeEntityFromDocumentEntity

File: src/main/java/com/google/sps/servlets/SearchServlet.java (hunk 11 above)
Comment (ramaroberto) [reply src/main/java/com/google/sps/servlets/SearchServlet.java]: Also

"String name = (String) recipeEntity.getProperty("title");"

Should be below the initialization of Recipe.

File: src/main/webapp/one-off/databaseInit.java [hunk 14]
```diff
@@ -40,104 +39,123 @@
...
 
-        datastore.put(recipeEntity);
-    }
+    Entity recipeEntity = new Entity(keyRange.getStart());
+    recipeEntity.setProperty("title", title);
+    recipeEntity.setProperty("index_title", title.toLowerCase());
+    recipeEntity.setProperty("imgURL", imgURL);
+    recipeEntity.setProperty("ingredients", ingredients);
+    recipeEntity.setProperty("stepList", stepList);
+    recipeEntity.setProperty("author", "Piece of Cake");
```
Comment (ramaroberto) [inline src/main/webapp/one-off/databaseInit.java]: Please explain in a comment why we're defining an ID here. Also modularize this into a function called buildRecipeEntity.

File: src/main/webapp/one-off/databaseInit.java [hunk 15]
```diff
@@ -40,104 +39,123 @@
...
-         List<String> lines = Collections.emptyList(); 
-         lines = Files.readAllLines(Paths.get(fileName), StandardCharsets.UTF_8); 
-         return lines; 
-    } 
+    Document recipeDocument = Document
+      .newBuilder()
+      .setId(String.valueOf(recipeEntity.getKey().getId()))
+      .addField(Field.newBuilder().setName("title").setText(title))
+      .addField(Field.newBuilder().setName("imgURL").setText(imgURL))
+      .build();
```
Comment (ramaroberto) [inline src/main/webapp/one-off/databaseInit.java]: You don't need the imgURL, also modularize this into a function called buildRecipeDocumentForIndexing

File: src/main/webapp/one-off/databaseInit.java [hunk 16]
```diff
@@ -40,104 +39,123 @@
...
-    
+    ArrayList<String> ingredients = new ArrayList<String>();
+    while (!StringUtils.isBlank(lines.get(currentLineIndex))) {
+      ingredients.add(lines.get(currentLineIndex));
+      currentLineIndex += 1;
     }
 
-   /**
-     * doGet loops trough all the 60 text files in the recipe folder an calls 'parseRecipe' on them
-    **/
```
Comment (ramaroberto) [inline src/main/webapp/one-off/databaseInit.java]: Why did you remove the comment?

File: src/main/webapp/one-off/databaseInit.java [hunk 17]
```diff
@@ -40,104 +39,123 @@
...
+  }
 
-       ArrayList<String> ingredients = new ArrayList<String>();
-       while (!lines.get(currentLineIndex).isBlank()){
-            ingredients.add(lines.get(currentLineIndex));
-            currentLineIndex += 1;
-       }
+  public void parseRecipe(String fileName) throws Exception, IOException {
+    List<String> lines = readFileInList(
+      "/home/donpaul/step/capstone/step249-2020/src/main/webapp/one-off/recipes/" +
```
Comment (ramaroberto) [inline src/main/webapp/one-off/databaseInit.java]: This should be defined in a constant outside the main class, or as a constant of the class.

RECIPES_DIRECTORY

File: src/main/webapp/one-off/databaseInit.java [hunk 18]
```diff
@@ -40,104 +39,123 @@
...
+    ArrayList<String> steps = new ArrayList<String>();
+    while (
+      (currentLineIndex < lines.size()) &&
+      (!StringUtils.isBlank(lines.get(currentLineIndex)))
+    ) {
+      steps.add(lines.get(currentLineIndex));
+      currentLineIndex += 1;
+    }
+    upload(title, imgURL, ingredients, steps);
+  }
```
Comment (ramaroberto) [inline src/main/webapp/one-off/databaseInit.java]: Would you mind adding comments about what each of these whiles are doing?

File: src/main/webapp/one-off/databaseInit.java [hunk 19]
```diff
@@ -40,104 +39,123 @@
...
-     for ( int i = 1; i <= 60; i++) {
-       String nameFile = Integer.toString(i)+".txt";
-       parseRecipe(nameFile);
-     }
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+    throws ServletException, IOException {
+    try {
+      for (int i = 1; i <= 60; i++) {
+        String nameFile = Integer.toString(i) + ".txt";
```
Comment (ramaroberto) [inline src/main/webapp/one-off/databaseInit.java]: The TXT extension string should be placed in a constant

RECIPE_FILE_EXTENSION

File: src/main/java/com/google/sps/servlets/Recipe.java (hunk 4 above)
Comment (donpaul999) [reply src/main/java/com/google/sps/servlets/Recipe.java]: I'll remove this, I used it only on the first version of the search (the inefficient one), with the edit distance.

File: src/main/java/com/google/sps/servlets/SearchServlet.java (hunk 7 above)
Comment (donpaul999) [reply src/main/java/com/google/sps/servlets/SearchServlet.java]: I was going to send the error value to the response, but this is not necessary anymore.

File: src/main/webapp/one-off/databaseInit.java (hunk 16 above)
Comment (donpaul999) [reply src/main/webapp/one-off/databaseInit.java]: I did it by mistake, I'll insert it back.

File: src/main/webapp/one-off/databaseInit.java (hunk 18 above)
Comment (donpaul999) [reply src/main/webapp/one-off/databaseInit.java]: No problem. I'll do this.

File: src/main/java/com/google/sps/servlets/Recipe.java [hunk 20]
```diff
@@ -1,9 +1,27 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
 package com.google.sps.servlets;
 
+/**
+ * Represents a Recipe
```
Comment (ramaroberto) [inline src/main/java/com/google/sps/servlets/Recipe.java]: period at the end of the sentence, same below

File: src/main/java/com/google/sps/servlets/SearchServlet.java [hunk 21]
```diff
@@ -26,25 +43,107 @@
...
 @WebServlet("/search")
 public class SearchServlet extends HttpServlet {
-  
-  /*
-  * doGet receives the request and returns the message sent as parameter and a hardcoded list of Recipes
-  */
+
+  /**
+   * Search and returns a list of first 10 recipes with the title matching the given parameter(searchterm).
+   * The index returns a list of documents in the ascending order by title
```
Comment (ramaroberto) [inline src/main/java/com/google/sps/servlets/SearchServlet.java]: Period at the end

File: src/main/java/com/google/sps/servlets/SearchServlet.java [hunk 22]
```diff
@@ -26,25 +43,107 @@
...
+      try {
+        Entity recipeEntity = getRecipeEntityFromDocumentEntity(
+          datastore,
+          entity
+        );
+        Recipe recipe = buildRecipe(recipeEntity);
+
+        recipesList.add(recipe);
+      } catch (EntityNotFoundException e) {
+        response.setStatus(505);
```
Comment (ramaroberto) [inline src/main/java/com/google/sps/servlets/SearchServlet.java:70]: You should do

 request.getRequestDispatcher("/search.jsp").forward(request, response);
return;

Here, you don't want to return any data if there was an error.

File: src/main/java/com/google/sps/servlets/SearchServlet.java [hunk 23]
```diff
@@ -26,25 +43,107 @@
...
+    String imgURL = (String) recipeEntity.getProperty("imgURL");
+    String description =
+      "Lorem quam dolor dapibus ante, sit amet pellentesque turpis lacus eu ipsum. Duis quis mi ut tortor interdum efficitur quis at mi. Pellentesque quis mauris vel ligula commodo scelerisque. In vulputate quam nisl, vel sagittis ipsum molestie quis. Suspendisse quis ipsum a sem aliquam euismod mattis sed metus.";
+
+    Recipe recipe = new Recipe();
+    recipe.setId(id);
+    recipe.setName(name);
+    recipe.setImage(imgURL);
+    recipe.setDescription(description);
+    return recipe;
```
Comment (ramaroberto) [inline src/main/java/com/google/sps/servlets/SearchServlet.java:135]: line break before this line

File: src/main/java/com/google/sps/servlets/SearchServlet.java [hunk 24]
```diff
@@ -26,25 +43,107 @@
...
+  private Entity getRecipeEntityFromDocumentEntity(
+    DatastoreService datastore,
+    ScoredDocument entity
+  )
+    throws EntityNotFoundException {
+    String id = entity.getId();
+    Entity recipeEntity = datastore.get(
+      KeyFactory.createKey("Recipe", Long.parseLong(id))
+    );
+    return recipeEntity;
```
Comment (ramaroberto) [inline src/main/java/com/google/sps/servlets/SearchServlet.java:148]: line break before this line

File: src/main/webapp/one-off/databaseInit.java [hunk 25]
```diff
@@ -40,104 +39,172 @@
...
-import java.nio.file.Paths;
-import java.nio.file.Files;
+import org.apache.commons.lang3.StringUtils;
 
-
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/test")
 public class TestUploadServlet extends HttpServlet {
+  private String RECIPES_DIRECTORY = "RECIPES_DIRECTORY";
+  private String RECIPE_FILE_EXTENSION = ".txt";
```
Comment (ramaroberto) [inline src/main/webapp/one-off/databaseInit.java]: You should use static or final:

https://www.thoughtco.com/constant-2034049

File: src/main/webapp/one-off/databaseInit.java [hunk 26]
```diff
@@ -40,104 +39,172 @@
...
-        recipeEntity.setProperty("author", "Piece of Cake");
-        recipeEntity.setProperty("difficulty", "N/A");
-        recipeEntity.setProperty("prep_time", "N/A");
-        recipeEntity.setProperty("cook_time", "N/A");
-
-        datastore.put(recipeEntity);
-    }
+     * We alocate an id here because we need it inside the index documents
+     * This will help us to search for the recipes in an efficient way
+     */
```
Comment (ramaroberto) [inline src/main/webapp/one-off/databaseInit.java]: These kind of comments are only used for documentation, use // comments here, even if they span multiple lines.

Also the comment is explaining the functionality but not the motivation of the id. It should say "The recipes are going to be accessed by the ID, henceforth we're reserving the ID here, setting it in the entity and saving it in the index. When we search for a recipe for any of its indexed attributes, the ID is going to be available for fetching the rest of the data from the datastore."

Also the ID here is not helping to "search the recipes in a efficient way", but to actually be able to load the data from the datastore for that recipe.

Try to think from the point of view of someone that's reading this piece of code for the first time, it should be possible to understand the WHY and HOW from the comment.

File: src/main/webapp/one-off/databaseInit.java [hunk 27]
```diff
@@ -40,104 +39,172 @@
...
+
+  /**
+   * Reads the file given as parameter as a list of strings
+   */
+
+  public static List<String> readFileInList(String fileName)
+    throws Exception, IOException {
+    List<String> lines = Collections.emptyList();
+    lines = Files.readAllLines(Paths.get(fileName), StandardCharsets.UTF_8);
+    return lines;
```
Comment (ramaroberto) [inline src/main/webapp/one-off/databaseInit.java]: Files.readAllLines is already returning a List<String>, you can do

return Files.readAllLines(Paths.get(fileName), StandardCharsets.UTF_8)

File: src/main/webapp/search.jsp [hunk 28]
```diff
@@ -1,21 +1,27 @@
...
-                        </p>
-                </li>
-            </c:forEach>
-        </ul>
+        <c:choose>
+            <c:when test="${recipesList.size() > 0}">
+                <ul id="results" class="list-group list-group-flush">
+                    <c:forEach items="${recipesList}" var="recipe">
+                        <li class="list-group-item">
+                            <img src="${recipe.getImage()}" class="img-thumbnail my-3 mr-3 float-left" style="width: 200px; height: 200px" alt="Recipe Image">
```
Comment (jtangelder) [inline src/main/webapp/search.jsp]: Please revert to use the `width` and `height` attributes instead of setting the style attribute.

**Review by jtangelder — APPROVED** [review]

File: src/main/webapp/one-off/databaseInit.java (hunk 26 above)
Comment (donpaul999) [reply src/main/webapp/one-off/databaseInit.java]: Ok, this is really helpful, thank you!

File: src/main/webapp/one-off/databaseInit.java [hunk 29]
```diff
@@ -40,104 +39,169 @@
...
+    throws Exception, InterruptedException, IOException {
+    IndexSpec indexSpec = IndexSpec
+      .newBuilder()
+      .setName("recipesIndex")
+      .build();
+    Index index = SearchServiceFactory.getSearchService().getIndex(indexSpec);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    //We allocate the ID before because the recipes are going to be acces by the ID later.
+    //The ID is set both to the entity, which is saved in the datastore, and to the document which is saved in the index.
```
Comment (ramaroberto) [inline src/main/webapp/one-off/databaseInit.java]: The comments should be

// Comment bla bla bla

There's an space after the "//". Also, please proofread everything even if it's coming from me.

And run the formatter, this is out of the 80 col limit!

File: src/main/webapp/one-off/databaseInit.java [hunk 30]
```diff
@@ -40,104 +39,169 @@
...
+    Document recipeDocument = buildRecipeDocumentForIndexing(recipeEntity);
+
+    datastore.put(recipeEntity);
+    index.put(recipeDocument);
+  }
+
+  /**
+   * Reads the file given as parameter as a list of strings.
+   */
+
```
Comment (ramaroberto) [inline src/main/webapp/one-off/databaseInit.java]: Extra breakline...

File: src/main/webapp/one-off/databaseInit.java [hunk 31]
```diff
@@ -40,104 +39,169 @@
...
-        recipeEntity.setProperty("difficulty", "N/A");
-        recipeEntity.setProperty("prep_time", "N/A");
-        recipeEntity.setProperty("cook_time", "N/A");
-
-        datastore.put(recipeEntity);
-    }
+  /**
+   * Receives a recipe's properties and creates an entity, for the database and a document for the index.
+   */
+  public void upload(
```
Comment (ramaroberto) [inline src/main/webapp/one-off/databaseInit.java]: Should be private

File: src/main/webapp/one-off/databaseInit.java [hunk 32]
```diff
@@ -40,104 +39,169 @@
...
+
+    datastore.put(recipeEntity);
+    index.put(recipeDocument);
+  }
+
+  /**
+   * Reads the file given as parameter as a list of strings.
+   */
+
+  public static List<String> readFileInList(String fileName)
```
Comment (ramaroberto) [inline src/main/webapp/one-off/databaseInit.java]: Should be private

File: src/main/webapp/one-off/databaseInit.java [hunk 33]
```diff
@@ -40,104 +39,169 @@
...
+    throws Exception, IOException {
+    return Files.readAllLines(Paths.get(fileName), StandardCharsets.UTF_8);
+  }
+
+  public void parseRecipe(String fileName) throws Exception, IOException {
```
Comment (ramaroberto) [inline src/main/webapp/one-off/databaseInit.java]: Should be private

**Review by ramaroberto — APPROVED** [review]
